### PR TITLE
Log error on DNSDone event.

### PR DESCRIPTION
**Pull request by tomwilkie**
Adding some more info to the HTTP tracing; helped my narrow down when my HTTP requests were intermittently failing.

### CODE REVIEW

File: nethttp/client.go [hunk 1]
```diff
@@ -242,8 +243,18 @@ func (h *Tracer) dnsStart(info httptrace.DNSStartInfo) {
...
+func (h *Tracer) dnsDone(info httptrace.DNSDoneInfo) {
+	fields := []log.Field{log.String("event", "DNSDone")}
+	if info.Err == nil {
+		addrs := make([]string, 0, len(info.Addrs))
+		for _, addr := range info.Addrs {
+			addrs = append(addrs, addr.String())
+		}
+		fields = append(fields, log.String("addrs", strings.Join(addrs, ",")))
+	} else {
+		fields = append(fields, log.String("error", info.Err.Error()))
```
Comment (yurishkuro) [inline nethttp/client.go]: log.Error(info.Err)

File: nethttp/client.go [hunk 2]
```diff
@@ -242,8 +243,18 @@ func (h *Tracer) dnsStart(info httptrace.DNSStartInfo) {
...
 }
 
-func (h *Tracer) dnsDone(httptrace.DNSDoneInfo) {
-	h.sp.LogFields(log.String("event", "DNSDone"))
+func (h *Tracer) dnsDone(info httptrace.DNSDoneInfo) {
+	fields := []log.Field{log.String("event", "DNSDone")}
+	if info.Err == nil {
+		addrs := make([]string, 0, len(info.Addrs))
+		for _, addr := range info.Addrs {
+			addrs = append(addrs, addr.String())
```
Comment (yurishkuro) [inline nethttp/client.go]: Why not add each address as its own log field? Fewer allocations.

**Comment by tomwilkie** [comment]
Thanks for the feedback!

Can you add multiple log fields with the same key to a span?

File: nethttp/client.go [hunk 3]
```diff
@@ -242,8 +242,14 @@ func (h *Tracer) dnsStart(info httptrace.DNSStartInfo) {
 	)
 }
 
-func (h *Tracer) dnsDone(httptrace.DNSDoneInfo) {
+func (h *Tracer) dnsDone(info httptrace.DNSDoneInfo) {
 	h.sp.LogFields(log.String("event", "DNSDone"))
```
Comment (yurishkuro) [inline nethttp/client.go]: I meant do a single span.LogFields() call with event=DNSDone, but append multiple `addr` fields to the slice.

If you're concerned with multiple fields having the same key (it's fine in Jaeger, but I guess the spec doesn't say exactly if it's allowed), then logging multiple events makes sense, but I would have `event=DNSDone` in all logs

Comment (tomwilkie) [reply nethttp/client.go]: Ah sorry I misunderstood..  should be fixed now.

**Comment by tomwilkie** [comment]
(travis failure seems to be due to travis misconfiguration...)

**Comment by yurishkuro** [comment]
@tomwilkie can you rebase off master? Travis build should be fixed now.

**Comment by tomwilkie** [comment]
Done!  Thanks.

**Pull request merged**